### PR TITLE
[Improve] Remove useless ReadonlyConfig flatten feature

### LINKED ISSUE #5597: [Bug] Support represent null in ReadonlyConfig
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/seatunnel/issues?q=is%3Aissue+label%3A%22bug%22) and found no similar issues.

### What happened

I find when I translate a Config to ReadonlyConfig, the null will be translate to "" in ReadonlyConfig.

This is due to below code.
image
e.g. If we have a : null in Config, it will be transform to "a=" in properties, then we split by '=', the value will be "".

### SeaTunnel Version

dev

### SeaTunnel Config

```conf
-
```

### Running Command

```shell
-
```

### Error Exception

```log
-
```

### Zeta or Flink or Spark Version

_No response_

### Java or Scala Version

_No response_

### Screenshots

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by Hisoka-X**
### Purpose of this pull request
This PR try to remove `flattern` in ReadonlyConfig.
Before this PR, as default behavior, ReadonlyConfig will try to flattern config key.
```hocon
// eg:
mapkey {
  inner {
   key = value
  }
}

// convert to
mapkey.inner.key = value
```
This feature seems unnecessary, and will affect the reading of other config that do not want flattern, and will cause errors when handling null values.

### Does this PR introduce _any_ user-facing change?
No.

### How was this patch tested?
exist test.

### Check list

* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update change log that in connector document. For more details you can refer to [connector-v2](https://github.com/apache/seatunnel/tree/dev/docs/en/connector-v2)
  2. Update [plugin-mapping.properties](https://github.com/apache/seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  3. Update the pom file of [seatunnel-dist](https://github.com/apache/seatunnel/blob/dev/seatunnel-dist/pom.xml)
* [ ] Update the [`release-note`](https://github.com/apache/seatunnel/blob/dev/release-note.md).

### CODE REVIEW

File: seatunnel-api/src/main/java/org/apache/seatunnel/api/configuration/ConfigAdapter.java [hunk 1]
```diff
@@ -31,9 +31,8 @@ public interface ConfigAdapter {
     String[] extensionIdentifiers();
 
     /**
-     * Converter config file to path_key-value Map (FlattenedMap) in HOCON
+     * Converter config file to path_key-value Map in HOCON
      *
-     * @see org.apache.seatunnel.api.configuration.util.ConfigUtil#flatteningMap(Map)
      * @param configFilePath config file path.
      * @return FlattenedMap
```
Comment (ruanwenjun) [inline seatunnel-api/src/main/java/org/apache/seatunnel/api/configuration/ConfigAdapter.java]:
```suggestion
     * @return Map
```

**Review by ruanwenjun — DISMISSED** [review]
LGTM

**Comment by ruanwenjun** [comment]
This PR will close #5597

**Review by hailin0 — COMMENTED** [review]
LGTM

**Review by hailin0 — APPROVED** [review]

**Review by ruanwenjun — APPROVED** [review]
+1

**Pull request merged**